### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by wb9688** [c9/core#328]
Is there a memory leak in it? By me, it uses a lot of memory (at the moment twice as much as Firefox) and every second it increases with ~10MB.

![Screenshot](https://cloud.githubusercontent.com/assets/9780201/16684559/f4c9128e-4505-11e6-8a4c-651a7c435b50.png)

**Comment by nightwing** [comment]
Yes, it looks like a leak, normally it should use around 20mb of memory.
What happens if you kill that process?
Does navigate work?

**Comment by wb9688** [comment]
What do you mean with "Does navigate work?"?

**Comment by nightwing** [comment]
the navigate panel, does it show any files.
From screenshot it is not visible which node process is taking all the memory, so if navigate panel doesn't work, it may be the file indexer.
Btw could you enable display of command line arguments in task manager, or use process explorer from https://technet.microsoft.com/en-us/sysinternals/processexplorer.aspx?f=255&MSPPError=-2147217396

**Comment by wb9688** [comment]
It works fine, but when I load a file or change to another tab, the console
shows a `ENOMEM` error (which means there isn't enough RAM left) and it'll
load  and display the file/tab and can edit the file. I'm currently not by
my laptop, so I can't check it with Process Explorer…

**Comment by wb9688** [comment]
In Process Explorer and in Task Manager it shows nothing under command line...

![Screenshot](https://cloud.githubusercontent.com/assets/9780201/16689226/447ba38e-4522-11e6-99d9-dbc55b53a84a.png)

**Comment by nightwing** [comment]
Are you using the new linux for window 10 to run this? If yes please try closing all terminals and restarting the server, to check if the leak is caused by pty.js

**Comment by wb9688** [comment]
Yes, I'm using Bash on Ubuntu on Windows. I've restarted my laptop mutliple
times…

**Comment by nightwing** [comment]
restarting the laptop is not related, if terminal is open it will leak the same memory again

**Comment by wb9688** [comment]
I thought you meant restarting the terminal as in restarting Bash on Ubuntu
on Windows. I'll try it tomorrow…

**Comment by nightwing** [comment]
It appears there are multiple issues running Cloud9 with Ubuntu on Windows. E.g. https://github.com/Microsoft/BashOnWindows/issues/118. I'd say it's better to wait for it to become more stable.
Until that the preferred way of running cloud9 is to use https://msys2.github.io/
